Gems/Atom/RHI/Code/Source/RHI/ResourceView.cpp

```diff
@@ -30,16 +30,41 @@ namespace AZ
             return ResultCode::Success;
         }
 
-        void ResourceView::Shutdown()
+        void ResourceView::release() const
         {
-            if (IsInitialized())
+            // It is possible that some other thread, not us, will delete this ResourceView after we
+            // decrement m_useCount. For example, another thread could create and release an instance
+            // immediately after we decrement. So we copy the necessary data to the callstack before
+            // decrementing. This ensures the call to ReleaseInstance() below won't crash even if this
+            // ResourceView gets deleted by another thread first.
+            ConstPtr<Resource> resource = m_resource;
+            const int useCount = --m_useCount;
+
+            AZ_Assert(useCount >= 0, "m_useCount is negative");
+
+            if (useCount == 0)
             {
-                ResourceInvalidateBus::Handler::BusDisconnect(m_resource.get());
-                ShutdownInternal();
-                
-                m_resource->EraseResourceView(this);
-                m_resource = nullptr;
-                DeviceObject::Shutdown();
+                if (IsInitialized())
```

Comment:
Maybe also force m_useCount to negative here.

Reply:
Here is where I don't think that would work, since another thread can always come along and increase the refcount, but that is expected behavior due to the ResourceView cache.